File: src/main/java/io/protobase/f7/formulas/logic/TRUE.java
package io.protobase.f7.formulas.logic;

import io.protobase.f7.formulas.AbstractFormula;
import io.protobase.f7.formulas.FormulaName;
import io.protobase.f7.models.GridColumnRowKey;

public class TRUE extends AbstractFormula {
  public static FormulaName NAME = FormulaName.TRUE;
  public static TRUE SELF = new TRUE();

  @Override
  public Object internal(GridColumnRowKey origin, Object... arguments) {
    checkLength(arguments.length, 0, NAME);
    return true;
  }
}

File: src/main/js/formulas/logic/TRUE.ts
import { SheetColumnRowKey } from "../../models/common/SheetColumnRowKey";
import { Complex } from "../../models/common/Types";
import { AbstractFormula } from "../AbstractFormula";
import { FormulaName } from "../FormulaName";

export class TRUE extends AbstractFormula {
  static SELF: TRUE = new TRUE();
  NAME = FormulaName.TRUE;

  internal(origin: SheetColumnRowKey, ...values: Array<Complex>) {
    AbstractFormula.checkLength(values.length, 0, this.NAME);
    return true;
  }
}
